components/action-mgt/org.wso2.carbon.identity.action.management/src/test/java/org/wso2/carbon/identity/action/management/ActionManagementServiceImplTest.java

```diff
@@ -0,0 +1,336 @@
+        Action deactivatedAction = serviceImpl.deactivateAction(
+                Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getPathParam(), action.getId(), tenantDomain);
+        Assert.assertEquals(Action.Status.INACTIVE, deactivatedAction.getStatus());
+    }
+
+    @Test(priority = 7)
+    public void testActivateAction() throws ActionMgtException {
+
+        Action result = serviceImpl.activateAction(
+                Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getPathParam(), action.getId(), tenantDomain);
+        Assert.assertEquals(Action.Status.ACTIVE, result.getStatus());
+    }
+
+    @Test(priority = 8)
+    public void testGetActionByActionId() throws ActionMgtException {
+
+        Action result = serviceImpl.getActionByActionId(action.getType().getPathParam(), action.getId(), tenantDomain);
+        Assert.assertEquals(action.getId(), result.getId());
+        Assert.assertEquals(action.getName(), result.getName());
+        Assert.assertEquals(action.getDescription(), result.getDescription());
+        Assert.assertEquals(action.getType(), result.getType());
+        Assert.assertEquals(action.getStatus(), result.getStatus());
+        Assert.assertEquals(action.getEndpoint().getUri(), result.getEndpoint().getUri());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getType(),
+                result.getEndpoint().getAuthentication().getType());
+    }
+
+    @Test(priority = 9)
+    public void testGetActionsCountPerType() throws ActionMgtException {
+
+        Map<String, Integer> actionMap = serviceImpl.getActionsCountPerType(tenantDomain);
+        for (Map.Entry<String, Integer> entry: actionMap.entrySet()) {
+            Assert.assertEquals(Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getActionType(), entry.getKey());
+            Assert.assertEquals(1, entry.getValue().intValue());
+        }
+    }
+
+    @Test(priority = 10)
+    public void testUpdateEndpointConfigWithSameAuth() throws ActionMgtException {
+
+        Authentication authType = buildMockBasicAuthentication("newadmin", "newadmin");
+        Action result = serviceImpl.updateActionEndpointAuthentication(
+                Action.ActionTypes.PRE_ISSUE_ACCESS_TOKEN.getPathParam(), action.getId(), authType, tenantDomain);
+        Assert.assertEquals(Authentication.Type.BASIC, result.getEndpoint().getAuthentication().getType());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getProperties().get(0).getValue(),
+                result.getEndpoint().getAuthentication().getProperties().get(0).getValue());
+        Assert.assertEquals(action.getEndpoint().getAuthentication().getProperties().get(1).getValue(),
+                result.getEndpoint().getAuthentication().getProperties().get(1).getValue());
+    }
+
+    @Test(priority = 11)
+    public void testUpdateEndpointConfigWithDifferentAuth() throws ActionMgtException {
+
+        Authentication authType = buildMockBearerAuthentication(ACCESS_TOKEN);
```

Comment:
```suggestion
        Authentication authentication = buildMockBearerAuthentication(ACCESS_TOKEN);
```
Since the object includes both type and the properties, let's use a more generic word

Reply:
Updated in